sssd_test_framework/utils/sbus.py

```diff
@@ -0,0 +1,459 @@
+"""Manage D-Bus operations"""
+
+from __future__ import annotations
+
+import re
+import xml.etree.ElementTree as xml
+from abc import ABC
+from enum import Enum, auto
+from typing import Any
+
+from pytest_mh import MultihostHost, MultihostUtility
+from pytest_mh.ssh import SSHProcessResult
+
+__all__ = [
+    "SBUSUtils",
+    "DBUSBus",
+    "Variant",
+]
+
+
+class DBUSBus(Enum):
+    """
+    Types of predefined D-Bus buses.
+    """
+
+    SYSTEM = auto()
+    """The system bus"""
+    SESSION = auto()
+    """The session bus"""
+
+
+class Variant:
+    """
+    Python representation for the D-Bus ``variant`` type.
+
+    This is a container for other types.
+    """
+
+    def __init__(self, value):
+        self.value = value
+        """The actual value stored in the variant type"""
+
+
+class DBUSObject(MultihostUtility[MultihostHost]):
+    """
+    D-Bus access.
+
+    Objects of this class represent the root or child D-Bus nodes.
+    """
+
+    class ProxyAttribute(ABC):
+        """
+        Abstract class representing a D-Bus attribute (method or property).
+
+        Corresponding concrete classes are provided for their instantiation.
+        """
+
+        def __init__(
+            self,
+            host: MultihostHost,
+            dest: str,
+            objpath: str,
+            bus: str | DBUSBus,
+            /,
+            interface: str | None = None,
+            child: xml.Element | None = None,
+        ):
+            match bus:
+                case DBUSBus.SYSTEM:
+                    _bus_param = "--system"
+                case DBUSBus.SESSION:
+                    _bus_param = "--session"
+                case _:
+                    _bus_param = "--bus=" + bus
+
+            self.host = host
+            self.dest = dest
+            self.objpath = objpath
+            self.bus = bus
+            self.interface = objpath[1:].replace("/", ".") if interface is None else interface
+            self.child = child
+            self.cmd = f"dbus-send {_bus_param} --print-reply --dest={dest} {self.objpath}"
+
+        @classmethod
+        def _get_dbus_type(cls, var):
+            """
+            Produce a basic python type from a text string describing a basic D-Bus type.
```

Comment:
This method does the opposite - produce string from python type.

Reply:
Gosh! Same thing for `_get_dbus_param()`. Fixed both.